src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -1156,34 +1165,50 @@ ReturnRowsFromTuplestore(uint64 tupleCount, TupleDesc tupleDescriptor,
 
 
 /*
- * GetConnectionForPlacement is the main entry point for acquiring a connection
- * within the router executor. By using placements (rather than node names and
- * ports) to identify connections, the router executor can keep track of shards
- * used by multi-statement transactions and error out if a transaction tries
- * to reach a new node altogether). In the single-statement commands context,
- * GetConnectionForPlacement simply falls through to  GetOrEstablishConnection.
+ * ManageShardIdParticipantForTransactionBlock for the given shardId and connection
+ * does the following:
+ *
+ * i)   if not in a transaction block, simply do nothing and return
+ * ii)  if the query is not modification and no modifications has happened
+ *      so far, do nothing
+ * iii) if the node has already been participated in a modification and the current
+ *      query is also a modification query, record the shard id participant
+ * iv)  if the query attempting to expand the affected worker nodes within the
```

Comment:
s/attempting/is attempting/
s/the affected worker nodes/the set of affected worker nodes/